### PR TITLE
Add Conan package management and CI/CD for multi-OS builds

### PR DESCRIPTION
**Pull request by damskii9992**
I would advise to do a squash and merge for this pull request due to the large amount of tiny commits in my development process.

### CODE REVIEW

File: CMakeUserPresets.json
Comment (anderskaestner) [inline CMakeUserPresets.json:1]: This is an absolute path on your system. Does it affect the process?

File: core/CMakeLists.txt
Comment (anderskaestner) [inline core/CMakeLists.txt:1]: Which python will be used here? There can be many pythons installed on the system (in anaconda, system, etc.) Is it still needed to specify the python on the cmake call?

File: core/algorithms/CMakeLists.txt
Comment (anderskaestner) [inline core/algorithms/CMakeLists.txt:1]: How often does the python need to be found? I guess this was a residue from my mess.

File: core/algorithms/UnitTests/tImagingAlgorithms/CMakeLists.txt
Comment (anderskaestner) [inline core/algorithms/UnitTests/tImagingAlgorithms/CMakeLists.txt:1]: I believe I can remove these commented blocks once all is confirmed and stable. Right?

File: core/algorithms/pyimgalg/CMakeLists.txt
Comment (anderskaestner) [inline core/algorithms/pyimgalg/CMakeLists.txt:1]: Nexus is currently deactivated in the code, but should be reactivated. Do you have a working Conan/cmake recipe for that?

File: profiles/macos_arm_clang_15_release
Comment (anderskaestner) [inline profiles/macos_arm_clang_15_release:1]: I presume this is the x-code compiler.

**Review by anderskaestner — COMMENTED** [review]
I have a few comments/questions on your PR. 
In general. I'm very grateful for the time you put into it. It looks a lot cleaner than before.

File: CMakeUserPresets.json
Comment (damskii9992) [reply CMakeUserPresets.json:1]: Oh, that should probably have been in .gitignore. My bad. It is created by the CMakeToolchain when running Conan. It does not affect anything and can be safely removed from the repo :)

File: core/CMakeLists.txt
Comment (damskii9992) [reply core/CMakeLists.txt:1]: In the build workflows the python version is managed by the Qt installation, which has a python version requirement range that currently resolves to 3.11.9. We can probably set this ourselves if we desire another python version in the Qt-allowed range of 3.6.x - 3.11.x, but it is generally a better idea to use the newest allowed version as is currently done.
Regarding which version the find_package call finds. On my system, I build locally in a conda environment, and CMake picks up the environments version of python, which is the intended behavior. So I don't specify anything special :)

**Pull request merged**